src/yunohost/data_migrations/0019_extends_permissions_features_1.py

```diff
@@ -0,0 +1,163 @@
+                })
+            else:
+                label = labels[permission.split('.')[0]].title()
+
+                if permission.endswith(".main"):
+                    ldap.update('cn=%s,ou=permission' % permission, {
+                        'authHeader': ["TRUE"],
+                        'label': [label],
+                        'showTile': ["TRUE"],
+                        'isProtected': ["FALSE"]
+                    })
+                else:
+                    ldap.update('cn=%s,ou=permission' % permission, {
+                        'authHeader': ["TRUE"],
+                        'label': ["%s (%s)" % (label, permission.split('.')[1])],
+                        'showTile': ["FALSE"],
+                        'isProtected': ["TRUE"]
+                    })
+
+    def migrate_skipped_unprotected_protected_uris(self, app=None):
+
+        logger.info(m18n.n("migration_0019_migrate_old_app_settings"))
+        apps = _installed_apps()
+
+        if app:
+            if app not in apps:
+                logger.error("Can't migrate permission for app %s because it ain't installed..." % app)
+                apps = []
+            else:
+                apps = [app]
+
+        def _get_setting(app, name):
+            s = app_setting(app, name)
+            return s.split(',') if s else []
+
+        for app in apps:
+            skipped_urls = [uri for uri in _get_setting(app, 'skipped_uris') if uri != '/']
+            skipped_urls += ['re:' + regex for regex in _get_setting(app, 'skipped_regex')]
+            unprotected_urls = [uri for uri in _get_setting(app, 'unprotected_uris') if uri != '/']
+            unprotected_urls += ['re:' + regex for regex in _get_setting(app, 'unprotected_regex')]
+            protected_urls = [uri for uri in _get_setting(app, 'protected_uris') if uri != '/']
+            protected_urls += ['re:' + regex for regex in _get_setting(app, 'protected_regex')]
+
+            if skipped_urls != []:
+                permission_create(app + ".legacy_skipped_uris", additional_urls=skipped_urls,
+                                  auth_header=False, label=legacy_permission_label(app, "skipped"),
+                                  show_tile=False, allowed='visitors', protected=True, sync_perm=False)
+            if unprotected_urls != []:
+                permission_create(app + ".legacy_unprotected_uris", additional_urls=unprotected_urls,
+                                  auth_header=True, label=legacy_permission_label(app, "unprotected"),
+                                  show_tile=False, allowed='visitors', protected=True, sync_perm=False)
+            if protected_urls != []:
+                permission_create(app + ".legacy_protected_uris", additional_urls=protected_urls,
+                                  auth_header=True, label=legacy_permission_label(app, "protected"),
+                                  show_tile=False, allowed=permission_list()['permissions'][app + ".main"]['allowed'],
```

Comment:
@Josue-T : Here I added a supposedly missing `[app+'.main']` otherwise that was probably miserably crashing ... can you confirm this was the initial intention ?

Reply:
Yes probably :+1: